src/test/java/games/strategy/engine/framework/startup/login/ClientLoginValidatorTests.java

```diff
@@ -15,26 +16,48 @@
 import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableMap;
 
 import games.strategy.engine.framework.startup.login.ClientLoginValidator.ErrorMessages;
-import games.strategy.net.IServerMessenger;
+import games.strategy.util.MD5Crypt;
+
 
 @RunWith(Enclosed.class)
 public final class ClientLoginValidatorTests {
   private static final String PASSWORD = "password";
   private static final String OTHER_PASSWORD = "otherPassword";
 
+  private static final String MAGIC_MAC_START = MD5Crypt.MAGIC + "MH$";
+
+  @RunWith(Enclosed.class)
+  public static final class MacValidation {
```

Comment:
Hmmm... I'm starting to question the enclosure framework we have a bit. @ssoloff , do you think it's still worthwhile having, or would we be better off keeping it simpler and having flat test files, even if maybe that means in some cases we have a 1:n between prod code and tests?

For me, the cases where you need `@RunWith(Enclosed.class)` are a bit nuanced, it makes me think we are doing something not terribly standard, why force everyone to learn yet another framework? I just want to be sure we are getting good value out of this.

The 'Test' vs 'Tests' suffix I do think is a bit wonky even, it's really not clear to the lay developer why/when you would need one or the other, it would be better IMO if we could focus on more core issues and not have test infrastructure intrude (I spent 5-10 minutes considering to rename classes, I was not even sure if the *Tests suffix would get picked up by standard tooling).

Reply:
Updated in: 60d9c33a7

Reply:
> would we be better off keeping it simpler and having flat test files, even if maybe that means in some cases we have a 1:n between prod code and tests?

Flat files are no good in cases like this because then you end up having fixture fields that are only used by subsets of test methods.  If nested fixtures are too distracting, I would recommend your second suggestion and allow for multiple test files per production class, where each file represents a specific fixture.

> For me, the cases where you need `@RunWith(Enclosed.class)` are a bit nuanced

I agree there's new rules to learn (which is why I offered it up for discussion in #1781 first :smile:), but not too many:

1. Non-leaf fixtures must be annotated with `@RunWith(Enclosed.class)` and be named with a `Tests` suffix.
1. Leaf fixtures must be named with a `Test` suffix (this requirement is no different than the non-nested case).
1. Nested fixtures (both leaf and non-leaf) must be declared `public static`.

As I said in #1781, I'm not married to `Enclosed` and would be ok with multiple test fixtures per production class (when multiple fixtures make sense).